src/Adapter/Product/Update/ProductStockUpdater.php

```diff
@@ -0,0 +1,188 @@
+        $this->stockManager = $stockManager;
+        $this->productRepository = $productRepository;
+        $this->stockAvailableRepository = $stockAvailableRepository;
+    }
+
+    /**
+     * @param Product $product
+     * @param array $propertiesToUpdate
+     * @param bool $addMovement
+     *
+     * @throws CoreException
+     * @throws ProductConstraintException
+     * @throws ProductPackConstraintException
+     * @throws ProductStockConstraintException
+     * @throws ProductStockException
+     */
+    public function update(Product $product, array $propertiesToUpdate, bool $addMovement = true): void
+    {
+        $stockAvailable = $this->getStockAvailable($product);
+        $this->productRepository->partialUpdate($product, $propertiesToUpdate, CannotUpdateProductException::FAILED_UPDATE_STOCK);
+
+        // It is very important to update StockAvailable after product, because the validation is performed in ProductRepository::partialUpdate
+        $this->updateStockAvailable($product, $stockAvailable, $propertiesToUpdate, $addMovement);
+
+        $advancedStockEnabled = (bool) $this->configuration->get('PS_ADVANCED_STOCK_MANAGEMENT');
+        if ($advancedStockEnabled && $product->depends_on_stock) {
+            StockAvailable::synchronize($product->id);
+        }
+    }
+
+    /**
+     * @param Product $product
+     * @param StockAvailable $stockAvailable
+     * @param array $propertiesToUpdate
+     * @param bool $addMovement
+     *
+     * @throws CoreException
+     * @throws ProductStockException
+     */
+    private function updateStockAvailable(Product $product, StockAvailable $stockAvailable, array $propertiesToUpdate, bool $addMovement): void
+    {
+        $stockUpdateRequired = false;
+        if (in_array('depends_on_stock', $propertiesToUpdate)) {
+            $stockAvailable->depends_on_stock = (bool) $product->depends_on_stock;
+            $stockUpdateRequired = true;
+        }
+        if (in_array('out_of_stock', $propertiesToUpdate)) {
+            $stockAvailable->out_of_stock = (int) $product->out_of_stock;
+            $stockUpdateRequired = true;
+        }
+        if (in_array('location', $propertiesToUpdate)) {
+            $stockAvailable->location = $product->location;
+            $stockUpdateRequired = true;
+        }
+
+        // Quantity is handled separately as it is also related to Stock movements
+        $stockUpdateRequired |= $this->updateQuantity($product, $stockAvailable, $propertiesToUpdate, $addMovement);
+
+        if ($stockUpdateRequired) {
+            $this->stockAvailableRepository->update($stockAvailable);
+        }
+    }
+
+    /**
+     * @param Product $product
+     * @param StockAvailable $stockAvailable
+     * @param array $propertiesToUpdate
+     * @param bool $addMovement
+     *
+     * @return bool
+     */
+    private function updateQuantity(Product $product, StockAvailable $stockAvailable, array $propertiesToUpdate, bool $addMovement): bool
```

Comment:
method name suits more for some `write` action rather than `read`.
maybe something like this? :
```suggestion
    private function isQtyUpdateRequired(Product $product, StockAvailable $stockAvailable, array $propertiesToUpdate, bool $addMovement): bool
```

Reply:
No because the method actually performs some updates (on the StockAvailable object, and in the StockMovement)
The result returned indicates if a field has been modified and if the object needs to be updated

Reply:
So it prevents from updating StockAvailable if no updated fields are related it (as the updater also handles configuration fields which are only related to Product and not StockAvailable) I'll add a bit more documentation though

Reply:
Ok I ended up modifying this function so that it doesn't return anything any more, the need for update is all handled in `updateStockAvailable`